core/xml.js

```diff
@@ -274,15 +274,21 @@ Blockly.Xml.textToDom = function(text) {
  * Decode an XML DOM and create blocks on the workspace.
  * @param {!Element} xml XML DOM.
  * @param {!Blockly.Workspace} workspace The workspace.
+ * @param {Array} offset array with x and y offset for the new blocks, default [0,0]
  */
-Blockly.Xml.domToWorkspace = function(xml, workspace) {
+Blockly.Xml.domToWorkspace = function(xml, workspace, offset) {
```

Comment:
I do know where the incoming blocks are located that's the second loop in the function which parse the xml file for the block locations.